nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -67,22 +67,38 @@ def __init__(self, cfg: DictConfig, trainer=None):
         vocabulary = self.tokenizer.tokenizer.get_vocab()
 
         # Set the new vocabulary
-        cfg.decoder.params.vocabulary = ListConfig(list(vocabulary.values()))
+        if "params" in cfg.decoder:
+            cfg.decoder.params.vocabulary = ListConfig(list(vocabulary.values()))
+        else:
+            cfg.decoder.vocabulary = ListConfig(list(vocabulary.values()))
 
         # Override number of classes if placeholder provided
-        if cfg.decoder.params['num_classes'] < 1:
+        if "params" in cfg.decoder:
+            num_classes = cfg.decoder["params"]["num_classes"]
+        else:
+            num_classes = cfg.decoder["num_classes"]
+
+        if num_classes < 1:
             logging.info(
                 "\nReplacing placeholder number of classes ({}) with actual number of classes - {}".format(
-                    cfg.decoder.params['num_classes'], len(vocabulary)
+                    num_classes, len(vocabulary)
                 )
             )
-            cfg.decoder.params['num_classes'] = len(vocabulary)
+            if "params" in cfg.decoder:
+                cfg.decoder["params"]["num_classes"] = len(vocabulary)
+            else:
+                cfg.decoder['num_classes'] = len(vocabulary)
```

Comment:
Does this work without OmegaConf's open_dict() or is this not an OmegaConf config?

Reply:
For this specific part, we assumed that num_classes is mandatory and always in the config file. Do we still need open_dict for here?

Reply:
I guess we don't then